homeassistant/components/onkyo/media_player.py

```diff
@@ -128,13 +128,27 @@
 
 type InputLibValue = str | tuple[str, ...]
 
-_cmds: dict[str, InputLibValue] = {
-    k: v["name"]
-    for k, v in {
-        **PYEISCP_COMMANDS["main"]["SLI"]["values"],
-        **PYEISCP_COMMANDS["zone2"]["SLZ"]["values"],
-    }.items()
-}
+
+def _input_lib_cmds(zone: str) -> dict[InputSource, InputLibValue]:
+    match zone:
+        case "main":
+            cmds = PYEISCP_COMMANDS["main"]["SLI"]
+        case "zone2":
+            cmds = PYEISCP_COMMANDS["zone2"]["SLZ"]
+        case "zone3":
+            cmds = PYEISCP_COMMANDS["zone3"]["SL3"]
+        case "zone4":
+            cmds = PYEISCP_COMMANDS["zone4"]["SL4"]
+
+    result: dict[InputSource, InputLibValue] = {}
+    for k, v in cmds["values"].items():
```

Comment:
For a future PR, shouldn't you cache the resulting dict as it's always the same ? You can do that by using the `@cache` decorator on thet `_input_lib_cmds` command .

Reply:
The reason I didn't is because this is called so rarely, that I didn't think it was needed. But I guess it shouldn't hurt either.